nemo/core/callbacks.py

```diff
@@ -37,6 +37,60 @@
 logging = nemo.logging
 
 
+class NeMoCallback(ABC):
+    def __init__(self):
+        self._action = None
+
+    @property
+    def action(self):
+        """TODO remove"""
+        return self._action
+
+    @action.setter
+    def action(self, action_obj):
+        self._action = action_obj
+
+    def on_action_start(self, state):
```

Comment:
I thought you proposed to have this set of "events"
on_train_start
on_epoch_start
on_optimizer_step_start
on_batch_start
on_batch_end
on_optimizer_step_stop
on_epoch_end
on_train_end

Reply:
I am against on_train_*, on_optimizer_* - there are specific for training action, and we already got two other types of **major actions** aside of training...

I would still suggest to use on_iteration_* instead of on_step_*, but I can live with it - as long we all agree on that name ;)